torch/distributed/_sharded_tensor/ops/equals.py

```diff
@@ -0,0 +1,51 @@
+import torch
+import torch.distributed as dist
+
+def _communicate_result(result, pg):
+    # Gather results from all ranks.
+    if result:
+        result_tensor = torch.ones(1, device=torch.cuda.current_device())
+    else:
+        result_tensor = torch.zeros(1, device=torch.cuda.current_device())
```

Comment:
If I read this correctly, if any of rank has result = False, this function will return `False` for sure. The purpose of this function is to ensure that if one rank gets a False result, all ranks will return `False`. Is my understanding correct?

Reply:
Yes your understanding is correct. In general we want all ranks to return True/False consistently. If we don't do this, some ranks might return True and other False which would be confusing.